### ISSUE TITLE
vagrant up hangs on "Waiting for SSH to become available"

### ISSUE THREAD

**Issue by shalako** [cloudfoundry/bosh-lite#355]
Spent two hours banging my head against this with no success.

```
$ vagrant up --provider=aws
Bringing machine 'default' up with 'aws' provider...
==> default: Checking if box 'cloudfoundry/bosh-lite' is up to date...
/Users/shannon/.vagrant.d/gems/gems/vagrant-aws-0.7.0/lib/vagrant-aws/action/run_instance.rb:98: warning: duplicated key at line 100 ignored: :associate_public_ip
==> default: Warning! The AWS provider doesn't support any of the Vagrant
==> default: high-level network configurations (`config.vm.network`). They
==> default: will be silently ignored.
==> default: Warning! You're launching this instance into a VPC without an
==> default: elastic IP. Please verify you're properly connected to a VPN so
==> default: you can access this machine, otherwise Vagrant will not be able
==> default: to SSH into it.
==> default: Launching an instance with the following settings...
==> default:  -- Type: m3.xlarge
==> default:  -- AMI: ami-5670703c
==> default:  -- Region: us-east-1
==> default:  -- Keypair: bosh
==> default:  -- Subnet ID: subnet-e34403de
==> default:  -- Security Groups: ["sg-0a9be072"]
==> default:  -- Block Device Mapping: [{:DeviceName=>"/dev/sda1", "Ebs.VolumeType"=>"gp2", "Ebs.VolumeSize"=>80}]
==> default:  -- Terminate On Shutdown: false
==> default:  -- Monitoring: false
==> default:  -- EBS optimized: false
==> default:  -- Source Destination check:
==> default:  -- Assigning a public IP address in a VPC: false
==> default:  -- VPC tenancy specification: default
==> default: Warning! Vagrant might not be able to SSH into the instance.
==> default: Please check your security groups settings.
==> default: Waiting for instance to become "ready"...
==> default: Waiting for SSH to become available...
```

If you CTL-C it deletes the instance!

[The instructions](https://github.com/cloudfoundry/bosh-lite/blob/master/docs/aws-provider.md) say "You can only deploy into a VPC if the instance can be accessed by the machine doing the deploying. If not, Vagrant will fail to use SSH to provision the instance further."

But the instructions do not say how Vagrant will fail (what is the expected error; I don't even know if this sentence applies to my problem), nor do the instructions say what to do about it.

I have tried:
- VPC configuration
  - DNS resolution: yes
  - DNS hostnames: yes
  
 With these set, the instance did not receive a hostname.
- Subnet configuration
  - Auto-assign Public IP: yes
  
 Only with this set did the instance received both a hostname and a public IP. However, it didn't help.

My security group is open to all.

**Comment by dpb587-pivotal** [comment]
This is typically caused by misconfigured security groups or subnets, or, in this case, not configuring vagrant to assign a public IP address and probably not using a VPN to access over private IP. Googling for the error message + "vagrant aws" points to https://github.com/mitchellh/vagrant-aws/issues/3.

Additionally, I think this particular topic was discussed and resolved in the slack channel.

Repo admin can probably close this.

**Comment by shalako** [comment]
Fixed by #356 - looking forward to that being pulled in.

**Comment by cppforlife** [comment]
#356 was merged. Thanks for docs updates!
